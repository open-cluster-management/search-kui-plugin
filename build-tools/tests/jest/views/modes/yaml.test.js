/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'
const yaml = require('../../../../../dist/src-web/views/modes/yaml')

const items = [
  {kind: require('../../../data/deployment')},
]
const deployment = items[0].kind.deployment

// Resource's YAML tab
describe('YAML mode', () => {
  const spy = jest.spyOn(yaml, 'yamlTab')
  const _ = yaml.yamlTab(deployment)

  it('should display an editor object with a detailed yaml', () => {
    expect(spy).toBeCalled()
    expect(yaml.yamlTab(deployment)).toMatchSnapshot()
  })
})
