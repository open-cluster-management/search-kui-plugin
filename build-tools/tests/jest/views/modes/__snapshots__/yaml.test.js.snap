// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YAML mode should display an editor object with a detailed yaml 1`] = `
Object {
  "content": "apiVersion: apps/v1
kind: Deployment
name: myapp
namespace: kube-system
cluster: local-cluster
selfLink: /api/v1/namespaces/kube-system/deployment/myapp
metadata:
  name: myapp
  namespace: kube-system
  labels:
    app: drone-app
spec:
  replicas: 1
  selector:
    matchLabels:
      app: drone-app
  template:
    metadata:
      labels:
        app: drone-app
    spec:
      terminationGracePeriodSeconds: 0
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            - labelSelector:
                matchExpressions:
                  - key: app
                    operator: In
                    values:
                      - drone-app
              topologyKey: kubernetes.io/hostname
      containers:
        - name: drone-app
          image: 'rvennam/drone-app:latest'
          ports:
            - containerPort: 3000
          resources:
            requests:
              memory: 64Mi
              cpu: 50m
            limits:
              memory: 128Mi
              cpu: 100m
",
  "contentType": "yaml",
  "label": "YAML",
  "mode": "yaml",
  "order": 2,
}
`;
